File: E_Commerce/E_Commerce/E_Commerce.Dal/Entities/Order.cs
using E_Commerce.Dal.Enums;

namespace E_Commerce.Dal.Entities;
public class Order
{
    public long OrderId { get; set; }
    public long CustomerId { get; set; }
    public DateTime CreatedAt { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal Discount { get; set; }
    public byte DiscountPercentage { get; set; }
    public decimal ServicePrice { get; set; }
    public Customer Customer { get; set; }
    public OrderStatus Status { get; set; }
    public List<OrderProduct> OrderProducts { get; set; }
    public List<Payment> Payments { get; set; }
}

File: Client/src/app/services/models/Order.ts
import {  CustomerModel } from "./Customer";
import {  OrderProductModel } from "./OrderProduct";
import {  PaymentModel } from "./Payment";

export enum OrderStatus {
  Pending = 'Created',
  Processing = 'Processing',
  Shipped = 'Shipped',
  Delivered = 'Delivered',
  Cancelled = 'Cancelled',
  // Add other statuses as needed
}

export class OrderModel {
  orderId: number = 0; // long in C# maps to number in TypeScript
  customerId: number = 0;
  createdAt: Date = new Date(); // DateTime in C# maps to Date in TypeScript
  totalAmount: number = 0; // decimal in C# maps to number in TypeScript
  discount: number = 0;
  discountPercentage: number = 0; // byte in C# maps to number in TypeScript
  servicePrice: number = 0;
  customer?: CustomerModel; // Optional, as it might be lazy-loaded or not always included
  status: OrderStatus = OrderStatus.Pending; // Using the defined OrderStatus enum
  orderProducts: OrderProductModel[] = []; // List<OrderProduct> in C# maps to OrderProduct[]
  payments: PaymentModel[] = []; // List<Payment> in C# maps to Payment[]
}
